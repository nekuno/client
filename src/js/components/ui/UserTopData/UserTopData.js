import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './UserTopData.scss';

export default class UserTopData extends Component {

    static propTypes = {
        username     : PropTypes.string.isRequired,
        usernameColor: PropTypes.string,
        location     : PropTypes.object,
        age          : PropTypes.string.isRequired,
        subColor     : PropTypes.string
    };

    render() {
        const {username, location, age, usernameColor, subColor} = this.props;

        return (
            <div className={styles.userTopData}>
                <div className={styles.username} style={{color: usernameColor}}>{username}</div>
                <div className={styles.ageCity} style={{color: subColor}}>{location ? location.locality || location.country || 'N/A' : 'N/A'} &bull; {age}</div>
            </div>
        );
    }
}

UserTopData.defaultProps = {
    usernameColor: 'white',
    subColor     : 'white'
};