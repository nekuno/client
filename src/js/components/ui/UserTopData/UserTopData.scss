@import '../../../../scss/_partials/colors';

.user-top-data {
  color: white;
  padding: 0 5%;
  .username {
    font-size: 20px;
  }
  .age-city {
    font-size: 14px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
}