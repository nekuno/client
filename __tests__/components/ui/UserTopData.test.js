import React from 'react';
import UserTopData from '../../../src/js/components/ui/UserTopData/UserTopData.js';
import styles from '../../../src/js/components/ui/UserTopData/UserTopData.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test UserTopData component', () => {
    const mockFn = jest.fn();
    const usertopdata = shallow(<UserTopData onClickHandler={mockFn}>Lorem ipsum</UserTopData>);

    it('should be defined', () => {
        expect(usertopdata).toBeDefined();
    });
    it('should render correctly', () => {
        expect(usertopdata).toMatchSnapshot();
    });
    it('UserTopData click event', () => {
        usertopdata.find(`.${styles.usertopdata}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
