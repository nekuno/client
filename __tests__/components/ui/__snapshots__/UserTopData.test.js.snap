// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UserTopData component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserTopData
    onClickHandler={[MockFunction]}
    subColor="white"
    usernameColor="white"
  >
    Lorem ipsum
  </UserTopData>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="username"
          style={
            Object {
              "color": "white",
            }
          }
        />,
        <div
          className="ageCity"
          style={
            Object {
              "color": "white",
            }
          }
        >
          N/A
           • 
        </div>,
      ],
      "className": "userTopData",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "username",
          "style": Object {
            "color": "white",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "N/A",
            " • ",
            undefined,
          ],
          "className": "ageCity",
          "style": Object {
            "color": "white",
          },
        },
        "ref": null,
        "rendered": Array [
          "N/A",
          " • ",
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="username"
            style={
              Object {
                "color": "white",
              }
            }
          />,
          <div
            className="ageCity"
            style={
              Object {
                "color": "white",
              }
            }
          >
            N/A
             • 
          </div>,
        ],
        "className": "userTopData",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "username",
            "style": Object {
              "color": "white",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "N/A",
              " • ",
              undefined,
            ],
            "className": "ageCity",
            "style": Object {
              "color": "white",
            },
          },
          "ref": null,
          "rendered": Array [
            "N/A",
            " • ",
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
